src/Concerns/CapturesState.php

```diff
@@ -110,7 +110,11 @@ public function configureGlobalRequestSampling(): void
      */
     public function configureGlobalCommandSampling(): void
     {
-        $this->sample($this->config['sampling']['commands']);
+        $this->sample(match (Compatibility::getSamplingFromContext(null)) {
+            true => 1.0,
+            false => 0.0,
+            null => $this->config['sampling']['commands'],
+        });
```

Comment:
Now this method uses not only the global sample rate but also the flag in the context. It might feel a bit confusing when the method is called `configureGlobalCommandSampling`.

This method could only configure the global sampling, and we could override the sampling only if the `nightwatch_should_sample` flag exists in the context.

What do you think?

Reply:
Yeah kinda. Are you thinking another method? Or rename?

Reply:
Otherwise, to further explain the current approach: it would exist in context already as a child process because the parent has scheduled the command and determined its sampling value to pass along, so it kinda is the global sample, its just derived from somewhere else

Reply:
Initially, I was thinking of another method like below, but it feels a bit weird to override the global sampling right after it's configured. Now I'm leaning towards renaming the method. e.g. `configureCommandSampling`

```php
private function registerCommandHooks(CommandStarting $event): void
{
    // ...

    $this->nightwatch->configureGlobalCommandSampling();

    $this->nightwatch->configureCommandSamplingFromContext();

    // ...
}
```

Reply:
I've renamed the methods now, I think that suits it a bit better